lib/spack/docs/build_settings.rst

```diff
@@ -209,6 +209,170 @@ Specific limitations include:
   then Spack will not add a new external entry (``spack config blame packages``
   can help locate all external entries).
 
+
+.. _manually-adding-external-packages:
+
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Manually Adding External Packages
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+This section discusses how to add external packages to the ``packages.yaml``
+file when modules depend on other modules or when Spack is unable to compute
+the prefix. When using modules, Spack will attempt to parse the ``module show``
+output in order to determine all relevant settings and variables. Reasons why
+this may not work automatically are
+
+* the wrong package name,
+* missing module dependencies,
+* nonstandard paths, and
+* the use of metamodules, i.e., modules whose only purpose is to load other
+  modules.
+
+For certain libraries there exist multiple implementations (e.g., MPI). If you
+modify the ``packages.yaml`` file, then you must make sure to select the
+correct package name. For example, the package ``mpich`` can be used only for
+the vanilla MPICH implementation but not for Cray MPICH; Cray MPICH has its own
+package called ``cray-mpich``. If the wrong module name is picked, this can
+cause errors later, e.g., Spack may compute a wrong prefix.
+
+The case of missing dependencies will be discussed based on the real-world
+example of loading OpenMPI 4.0.2 with CUDA support. Shown below is the ``module
+show openmpi/4.0.2`` output.
+
+.. code-block:: console
+
+    $ module show openmpi/4.0.2-cuda
+    -------------------------------------------------------------------
+    /gpfslocalsup/pub/modules-idris-env4/modulefiles/linux-rhel8-skylake_avx512/openmpi/4.0.2-cuda:
+
+    module-whatis   {An open source Message Passing Interface implementation.}
+    prereq          intel-compilers/19.0.4 pgi/20.1 pgi/19.10 gcc/10.1.0 gcc/8.3.1
```

Comment:
As a side note: we don't currently search through modules at all to detect new packages (e.g. `module avail`), as mentioned above (in this file) at https://spack.readthedocs.io/en/latest/build_settings.html?highlight=external#automatically-find-external-packages. If we add that, we should probably consider checking for prereq modules and trying to load them first.